commit 66cbbebcc616db8e221e81f7ecbb2c77701d031b
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Thu May 10 15:17:03 2012 -0700

    Hurd: Fix mlock in all cases except non-readable pages.

---
 sysdeps/mach/hurd/mlock.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/sysdeps/mach/hurd/mlock.c
+++ b/sysdeps/mach/hurd/mlock.c
@@ -1,5 +1,5 @@
 /* mlock -- guarantee pages are resident in memory.  Mach/Hurd version.
-   Copyright (C) 2001 Free Software Foundation, Inc.
+   Copyright (C) 2001-2012 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -40,7 +40,7 @@
   page = trunc_page ((vm_address_t) addr);
   len = round_page ((vm_address_t) addr + len) - page;
   err = __vm_wire (hostpriv, __mach_task_self (), page, len,
-		   VM_PROT_ALL); /* XXX ? */
+		   VM_PROT_READ);
   __mach_port_deallocate (__mach_task_self (), hostpriv);
 
   return err ? __hurd_fail (err) : 0;
